File: library/src/main/java/name/zeno/android/presenter/items/TextItem.java
package name.zeno.android.presenter.items;

import android.view.View;
import android.widget.TextView;

import kale.adapter.item.AdapterItem;

/**
 * @author 陈治谋 (513500085@qq.com)
 * @since 2016/12/5.
 */
public class TextItem implements AdapterItem<String>
{
  private TextView root;

  @Override public int getLayoutResId()
  {
    return android.R.layout.simple_list_item_1;
  }

  @Override public void bindViews(View root)
  {
    this.root = (TextView) root;

  }

  @Override public void setViews()
  {

  }

  @Override public void handleData(String s, int position)
  {
    root.setText(s);
  }
}

File: library/src/main/java/name/zeno/android/presenter/items/TextItem.kt
package name.zeno.android.presenter.items

import android.view.View
import android.widget.TextView

import kale.adapter.item.AdapterItem

/**
 * @author 陈治谋 (513500085@qq.com)
 * @since 2016/12/5.
 */
@Suppress("unused")
class TextItem : AdapterItem<String> {
  private lateinit var root: TextView
  override fun getLayoutResId(): Int = android.R.layout.simple_list_item_1

  override fun bindViews(root: View) {
    this.root = root as TextView
  }

  override fun setViews() {}

  override fun handleData(text: String, position: Int) {
    root.text = text
  }
}
